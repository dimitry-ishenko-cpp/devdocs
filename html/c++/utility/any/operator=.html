    <h1 id="firstHeading" class="firstHeading">std::any::operator=</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">any&amp; operator=( const any&amp; rhs );</pre>
</td> <td> (1) </td> <td> <span class="t-mark-rev t-since-cxx17">(since C++17)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">any&amp; operator=( any&amp;&amp; rhs ) noexcept;</pre>
</td> <td> (2) </td> <td> <span class="t-mark-rev t-since-cxx17">(since C++17)</span> </td> </tr> <tr class="t-dcl t-since-cxx17"> <td> <pre data-language="cpp">template&lt; typename ValueType &gt;
    any&amp; operator=( ValueType&amp;&amp; rhs );</pre>
</td> <td> (3) </td> <td> <span class="t-mark-rev t-since-cxx17">(since C++17)</span> </td> </tr>  </table> <p>Assigns contents to the contained value.</p>
<div class="t-li1">
<span class="t-li">1)</span> Assigns by copying the state of <code>rhs</code>, as if by <code>any(rhs).swap(*this)</code>.</div> <div class="t-li1">
<span class="t-li">2)</span> Assigns by moving the state of <code>rhs</code>, as if by <code>any(std::move(rhs)).swap(*this)</code>. <code>rhs</code> is left in a valid but unspecified state after the assignment.</div> <div class="t-li1">
<span class="t-li">3)</span> Assigns the type and value of <code>rhs</code>, as if by <code>any<span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>ValueType<span class="sy1">&gt;</span><span class="br0">(</span>rhs<span class="br0">)</span><span class="br0">)</span>.<span class="me1">swap</span><span class="br0">(</span><span class="sy2">*</span>this<span class="br0">)</span></code>. This overload participates in overload resolution only if <code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">std::<span class="me2">decay_t</span></span></a><span class="sy1">&lt;</span>ValueType<span class="sy1">&gt;</span></code> is not the same type as <code>any</code> and <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">std::<span class="me2">is_copy_constructible_v</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">std::<span class="me2">decay_t</span></span></a><span class="sy1">&lt;</span>ValueType<span class="sy1">&gt;&gt;</span></code> is <code>true</code>.</div>  <h3 id="Template_parameters"> Template parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> ValueType </td> <td> - </td> <td> contained value type </td>
</tr> <tr class="t-par-hitem"> <td colspan="3"> Type requirements </td>
</tr> <tr class="t-par-req"> <td colspan="3"> -<code><a href="http://en.cppreference.com/w/cpp/types/decay"><span class="kw632">std::<span class="me2">decay_t</span></span></a><span class="sy1">&lt;</span>ValueType<span class="sy1">&gt;</span></code> must meet the requirements of <a href="../../named_req/copyconstructible.html" title="cpp/named req/CopyConstructible">CopyConstructible</a>. </td>
</tr>
</table> <h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> rhs </td> <td> - </td> <td> object whose contained value to assign </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p><code>*this</code></p>
<h3 id="Exceptions"> Exceptions</h3> <div class="t-li1">
<span class="t-li">1,3)</span> Throws <code><a href="../../memory/new/bad_alloc.html" title="cpp/memory/new/bad alloc">std::bad_alloc</a></code> or any exception thrown by the constructor of the contained type. If an exception is thrown, there are no effects (strong exception guarantee).</div> <h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="any.html" title="cpp/utility/any/any"> <span class="t-lines"><span>(constructor)</span></span></a></div> </td> <td> constructs an <code>any</code> object <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/any/operator%3D" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/any/operator%3D</a>
  </p>
</div>
