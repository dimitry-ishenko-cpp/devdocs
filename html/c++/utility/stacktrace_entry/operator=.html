    <h1 id="firstHeading" class="firstHeading">std::stacktrace_entry::operator=</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr stacktrace_entry&amp; operator=( const stacktrace_entry&amp; other ) noexcept;</pre>
</td> <td class="t-dcl-nopad"> </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr>  </table> <p>Copy assignment operator. Replaces the contents of <code>*this</code> with those of <code>other</code>.</p>
<h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> other </td> <td> - </td> <td> another <code>stacktrace_entry</code> to assign from </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p><code>*this</code>.</p>
<div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/stacktrace_entry/operator%3D" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/stacktrace_entry/operator%3D</a>
  </p>
</div>
