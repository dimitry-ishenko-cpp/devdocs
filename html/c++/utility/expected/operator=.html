    <h1 id="firstHeading" class="firstHeading">std::expected&lt;T,E&gt;::operator=</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">constexpr expected&amp; operator=( const expected&amp; other );</pre></td> <td> (1)</td> <td><span class="t-mark-rev t-since-cxx23">(since C++23)</span></td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">constexpr expected&amp; operator=( expected&amp;&amp; other ) noexcept(/*see below*/);</pre></td> <td> (2)</td> <td><span class="t-mark-rev t-since-cxx23">(since C++23)</span></td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">template&lt; class U = T &gt;
constexpr expected&amp; operator=( U&amp;&amp; v );</pre></td> <td> (3)</td> <td>
<span class="t-mark-rev t-since-cxx23">(since C++23)</span> <br><span class="t-mark">(T is not <i>cv</i> void)</span>
</td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">template&lt; class G &gt;
constexpr expected&amp; operator=( const unexpected&lt;G&gt;&amp; other );</pre></td> <td> (4)</td> <td><span class="t-mark-rev t-since-cxx23">(since C++23)</span></td> </tr> <tr class="t-dcl t-since-cxx23"> <td><pre data-language="cpp">template&lt; class G &gt;
constexpr expected&amp; operator=( unexpected&lt;G&gt;&amp;&amp; other );</pre></td> <td> (5)</td> <td><span class="t-mark-rev t-since-cxx23">(since C++23)</span></td> </tr>  </table> <p>Assigns a new value to an existing <code>expected</code> object.</p>
<span class="t-li">1,2)</span> Assigns the state of <code>other</code>. <ul>
<li> If <a href="operator_bool.html" title="cpp/utility/expected/operator bool"><code>this-&gt;has_value()</code></a> equals <code>other.has_value()</code>, assigns the value contained in <code>other</code>. Does nothing if <code>T</code> is (possibly cv-qualified) <code>void</code> and <code>other.has_value()</code> is <span class="kw2">true</span>. </li>
<li> Otherwise, destroys the currently contained value (does nothing if <code>this-&gt;has_value()</code> is <span class="kw2">true</span> and <code>T</code> is (possibly cv-qualified) <code>void</code>), and makes <code>*this</code> contain a copy of the value contained in <code>other</code>. </li>
</ul>  If <code>other.has_value()</code> is <span class="kw2">true</span> and <code>T</code> is (possibly cv-qualified) <code>void</code>, does not construct the new value. Otherwise, the new value is copy-constructed <span class="t-v">(<a href="#Version_1">1</a>)</span> or move-constructed <span class="t-v">(<a href="#Version_2">2</a>)</span> from <code>*other</code> or <code>other.error()</code>, as appropriate. If an exception is thrown, the old value is retained; <code>*this</code> does not become valueless.  <p>If no exception was thrown, after assignment, <code>has_value()</code> is equal to <code>other.has_value()</code>.</p>
<ul>
<li> Overload <span class="t-v">(<a href="#Version_1">1</a>)</span> is defined as deleted unless <ul>
<li> either <code>T</code> is (possibly cv-qualified) <code>void</code> or <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_assignable"><span class="kw558">std::<span class="me2">is_copy_assignable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> either <code>T</code> is (possibly cv-qualified) <code>void</code> or <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">std::<span class="me2">is_copy_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_assignable"><span class="kw558">std::<span class="me2">is_copy_assignable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_copy_constructible"><span class="kw540">std::<span class="me2">is_copy_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> at least one of the following is <span class="kw2">true</span>: <ul>
<li> <code>T</code> is (possibly cv-qualified) <code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> </li>
</ul> </li>
</ul> </li>
<li> Overload <span class="t-v">(<a href="#Version_2">2</a>)</span> participates in overload resolution only if <ul>
<li> either <code>T</code> is (possibly cv-qualified) <code>void</code> or <code><a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw564">std::<span class="me2">is_move_assignable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> either <code>T</code> is (possibly cv-qualified) <code>void</code> or <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">std::<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw564">std::<span class="me2">is_move_assignable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">std::<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> at least one of the following is <span class="kw2">true</span>: <ul>
<li> <code>T</code> is (possibly cv-qualified) <code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code>
</li>
</ul>
</li>
</ul>
</li>
</ul> <span class="t-li">3)</span> Assigns from expected value. <ul>
<li> If <code>this-&gt;has_value()</code> is <span class="kw2">true</span>, equivalent to <code><span class="sy2">**</span>this <span class="sy1">=</span> <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span><span class="br0">(</span>v<span class="br0">)</span></code>. </li>
<li> Otherwise, destroys the value contained in <code>*this</code>, and makes <code>*this</code> contain a value initialized from <code><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span><span class="br0">(</span>v<span class="br0">)</span></code>. If an exception is thrown, the old value is retained; <code>*this</code> does not become valueless. </li>
</ul> <p>If no exception was thrown, after assignment, <code>this-&gt;has_value()</code> is <span class="kw2">true</span>.</p>
<ul><li> This overload participates in overload resolution only if <ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_same"><span class="kw592">std::<span class="me2">is_same_v</span></span></a><span class="sy1">&lt;</span>expected, <a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">std::<span class="me2">remove_cvref_t</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;&gt;</span></code> is false, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">std::<span class="me2">remove_cvref_t</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code> is not a specialization of <code>std::unexpected</code>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">std::<span class="me2">is_constructible_v</span></span></a><span class="sy1">&lt;</span>T, U<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_assignable"><span class="kw552">std::<span class="me2">is_assignable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy3">&amp;</span>, U<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> at least one of the following is <span class="kw2">true</span>: <ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw532">std::<span class="me2">is_nothrow_constructible_v</span></span></a><span class="sy1">&lt;</span>T, U<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code>
</li>
</ul>
</li>
</ul>
</li></ul> <span class="t-li">4,5)</span> Assigns from unexpected value. <p>Let <code>GF</code> be <code>const G&amp;</code> for overload <span class="t-v">(<a href="#Version_4">4</a>)</span>, and <code>G</code> for overload <span class="t-v">(<a href="#Version_5">5</a>)</span>.</p>
<ul>
<li> If <code>this-&gt;has_value()</code> is <span class="kw2">true</span>, destroys the value contained in <code>*this</code> (does nothing if <code>T</code> is (possibly cv-qualified) <code>void</code>), and makes <code>*this</code> contain a value initialized from <code><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>GF<span class="sy1">&gt;</span><span class="br0">(</span>e.<span class="me1">error</span><span class="br0">(</span><span class="br0">)</span><span class="br0">)</span></code>. If an exception is thrown, the old value is retained; <code>*this</code> does not become valueless. </li>
<li> Otherwise, equivalent to <code>this<span class="sy2">-</span><span class="sy1">&gt;</span>error<span class="br0">(</span><span class="br0">)</span> <span class="sy1">=</span> <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>GF<span class="sy1">&gt;</span><span class="br0">(</span>e.<span class="me1">error</span><span class="br0">(</span><span class="br0">)</span><span class="br0">)</span></code>. </li>
</ul> <p>If no exception was thrown, after assignment, <code>this-&gt;has_value()</code> is false.</p>
<ul><li> This overload participates in overload resolution only if <ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">std::<span class="me2">is_constructible_v</span></span></a><span class="sy1">&lt;</span>E, GF<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_assignable"><span class="kw552">std::<span class="me2">is_assignable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy3">&amp;</span>, GF<span class="sy1">&gt;</span></code> is <span class="kw2">true</span>, and </li>
<li> at least one of the following is <span class="kw2">true</span>: <ul>
<li> <code>T</code> is (possibly cv-qualified) <code>void</code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw532">std::<span class="me2">is_nothrow_constructible_v</span></span></a><span class="sy1">&lt;</span>E, GF<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span></code> </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span></code>
</li>
</ul>
</li>
</ul>
</li></ul> <p><br> In all cases, if <code>T</code> is not (possibly cv-qualified) <code>void</code>, the destruction of old value and construction of new value is performed as if by the following exposition-only function <code>reinit_expected</code>.</p>
<div class="cpp source-cpp"><pre data-language="cpp">template&lt;class NewType, class OldType, class... Args&gt;
constexpr void reinit_expected(NewType&amp; new_val, OldType&amp; old_val, Args&amp;&amp;... args)
{
    if constexpr (std::is_nothrow_constructible_v&lt;NewType, Args...&gt;)
    {
        std::destroy_at(std::addressof(old_val));
        std::construct_at(std::addressof(new_val), std::forward&lt;Args&gt;(args)...);
    }
    else if constexpr (std::is_nothrow_move_constructible_v&lt;NewType&gt;)
    {
        NewType temp(std::forward&lt;Args&gt;(args)...);
        std::destroy_at(std::addressof(old_val));
        std::construct_at(std::addressof(new_val), std::move(temp));
    }
    else
    {
        OldType temp(std::move(old_val));
        std::destroy_at(std::addressof(old_val));
        try
        {
            std::construct_at(std::addressof(new_val), std::forward&lt;Args&gt;(args)...);
        }
        catch (...)
        {
            std::construct_at(std::addressof(old_val), std::move(temp));
            throw;
        }
    }
}</pre></div>  <h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> other </td> <td> - </td> <td> another <code>expected</code> object whose contained value to assign </td>
</tr> <tr class="t-par"> <td> value </td> <td> - </td> <td> value to assign to the contained value </td>
</tr> <tr class="t-par"> <td> e </td> <td> - </td> <td> <a href="unexpected.html" title="cpp/utility/expected/unexpected"><code>std::unexpected</code></a> object whose contained value to assign </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p><code>*this</code></p>
<h3 id="Exceptions"> Exceptions</h3> <div class="t-li1">
<span class="t-li">1)</span> Throws any exception thrown by the copy constructor or copy assignment operator of <code>T</code> or <code>E</code>.</div> <div class="t-li1">
<span class="t-li">2)</span> If <code>T</code> is (possibly cv-qualified) <code>void</code>, <div class="t-noexcept-full">
<a href="../../language/noexcept_spec.html" title="cpp/language/noexcept spec"><code>noexcept</code></a> specification: <div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">noexcept</span><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw568">std::<span class="me2">is_nothrow_move_assignable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span><span class="br0">)</span></code></span></div>
</div> Otherwise, <div class="t-noexcept-full">
<a href="../../language/noexcept_spec.html" title="cpp/language/noexcept spec"><code>noexcept</code></a> specification: <div class="t-noexcept-box"><span class="t-cc"><code><span class="kw1">noexcept</span><span class="br0">(</span><br> <p><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw568">std::<span class="me2">is_nothrow_move_assignable_v</span></span></a><span class="sy1">&lt;</span>T<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span><br>  <a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw550">std::<span class="me2">is_nothrow_move_constructible_v</span></span></a></p>
<span class="sy1">&lt;</span>E<span class="sy1">&gt;</span> <span class="sy3">&amp;&amp;</span> <a href="http://en.cppreference.com/w/cpp/types/is_move_assignable"><span class="kw568">std::<span class="me2">is_nothrow_move_assignable_v</span></span></a><span class="sy1">&lt;</span>E<span class="sy1">&gt;</span><span class="br0">)</span></code></span></div>
</div>
</div> <div class="t-li1">
<span class="t-li">3)</span> Throws any exception thrown by the constructor or assignment operator of <code>T</code>.</div> <div class="t-li1">
<span class="t-li">4,5)</span> Throws any exception thrown by the constructor or assignment operator of <code>E</code>.</div> <h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="emplace.html" title="cpp/utility/expected/emplace"> <span class="t-lines"><span>emplace</span></span></a></div> </td> <td> constructs the expected value in-place <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/expected/operator%3D" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/expected/operator%3D</a>
  </p>
</div>
